### PR TITLE
Handle collection updates

### PR DESCRIPTION
**Pull request by sanderpick**
Small one here... catches invalid schema errors and force updates the collection.

### CODE REVIEW

File: buckets/buckets.go [hunk 1]
```diff
@@ -212,7 +212,24 @@ func (b *Buckets) Save(ctx context.Context, dbID thread.ID, bucket *Bucket, opts
...
+				return err
+			}
+			return b.Save(ctx, dbID, bucket, opts...)
+		}
+		return err
+	}
+	return nil
+}
+
+func ensureNoNulls(b *Bucket) {
```
Comment (sanderpick) [inline buckets/buckets.go:229]: We'll have to continue to stack field migrations here until there's a better way (see https://github.com/textileio/go-threads/issues/363)

File: api/buckets/service.go [hunk 2]
```diff
@@ -553,7 +553,13 @@ func (s *Service) RemovePath(ctx context.Context, req *pb.RemovePathRequest) (*p
 		return nil, err
 	}
 	if err = s.IPFSClient.Pin().Update(ctx, buckPath, dirpth); err != nil {
-		return nil, err
+		if err.Error() == "'from' cid was not recursively pinned already" {
```
Comment (jsign) [inline api/buckets/service.go:556]: Ouch! :(

File: buckets/buckets.go [hunk 3]
```diff
@@ -212,7 +212,24 @@ func (b *Buckets) Save(ctx context.Context, dbID thread.ID, bucket *Bucket, opts
...
-	return b.threads.Save(ctx, dbID, CollectionName, dbc.Instances{bucket}, db.WithTxnToken(args.Token))
+	ensureNoNulls(bucket)
+	err := b.threads.Save(ctx, dbID, CollectionName, dbc.Instances{bucket}, db.WithTxnToken(args.Token))
+	if err != nil {
+		if isInvalidSchemaErr(err) {
+			if err := b.updateCollection(ctx, dbID, opts...); err != nil {
+				return err
+			}
+			return b.Save(ctx, dbID, bucket, opts...)
+		}
```
Comment (jsign) [inline buckets/buckets.go:223]: Maybe make `isInvalidSchemaErr` do the check `err != nil` and avoid some nesting?
Like:
```go
if isInvalidSchemaErr(err) {
   ...
}
if err != nil {
   return fmt.Errorf("saving bucket in thread: %s", err)
}
return nil

Comment (sanderpick) [reply buckets/buckets.go:223]: Sure, can do

File: api/buckets/service.go (hunk 2 above)
Comment (sanderpick) [reply api/buckets/service.go:556]: No big deal, this was only reachable by encountering bucket save errors from a schema mismatch... so, shouldn't have happened to anyone here.

**Pull request merged**